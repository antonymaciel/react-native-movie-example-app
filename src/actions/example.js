

import Api from "../api/apiMovies"

const action = () => {
    Api.getMovieReviews('429617')
    .then(response => 
        console.log('RESPONSE', response)
    )
    .catch(err => {
    console.log(err)
    })
    return 'hola'
  }
 