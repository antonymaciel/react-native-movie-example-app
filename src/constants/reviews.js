export const REVIEWS_SUCCESS = 'REVIEWS_SUCCESS';
export const REVIEWS_FAILD = 'REVIEWS_FAILD';
export const REVIEWS_LOADING = 'REVIEWS_LOADING';
export const REVIEWS_FINISH = 'REVIEWS_FINISH';
