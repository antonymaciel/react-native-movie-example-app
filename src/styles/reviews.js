import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
    title: {
        fontSize: 20,
        fontWeight: '500'
    },
    headerContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        margin: 20,
    },
    image: {
        marginRight: 40,
        height: 130,
        width: 90
    },
    reviewList: {
        flex: 1
    }
});


export default styles;
