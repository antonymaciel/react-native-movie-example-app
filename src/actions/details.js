import Api from "../api/apiMovies"
import { DETAILS_SUCCESS, DETAILS_FAILD, DETAILS_LOADING, DETAILS_FINISH } from '../constants/details'

const movieDetailsSuccess = (movieDetails) => ({
    type: DETAILS_SUCCESS,
    movieDetails
});

const movieDetailsFaild = () => ({
    type: DETAILS_FAILD, 
});

const movieDetailsLoading = () => ({
    type: DETAILS_LOADING,
});

export const movieDetailsFinish = () => ({
    type: DETAILS_FINISH,
});

export const getMovieDetails = (idMovie) => (dispatch) => {
    dispatch(movieDetailsLoading());
    Api.getMovieDetails(idMovie)
    .then((json) => {
        dispatch(movieDetailsSuccess(json));
    })
    .catch(err => {
        console.log(err);
        dispatch(movieDetailsFaild());
    })
  };