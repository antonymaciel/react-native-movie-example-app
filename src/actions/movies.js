import Api from "../api/apiMovies"
import { MOVIES_SUCCESS, MOVIES_FAILD, MOVIES_LOADING } from '../constants/movies'

export const moviesSuccess = (movies) => ({
    type: MOVIES_SUCCESS,
    movies 
});

export const moviesFaild = () => ({
    type: MOVIES_FAILD, 
});

export const moviesLoading = () => ({
    type: MOVIES_LOADING,
});

export const getMovies = (page) => (dispatch) => {
    dispatch(moviesLoading());
    Api.getMovies(page)
    .then((json) => {
        dispatch(moviesSuccess(json));
    })
    .catch(err => {
        console.log(err);
        dispatch(moviesFaild());
    })
  };
 