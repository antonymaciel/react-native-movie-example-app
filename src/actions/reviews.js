import Api from "../api/apiMovies"
import { REVIEWS_SUCCESS, REVIEWS_FAILD, REVIEWS_LOADING, REVIEWS_FINISH } from '../constants/reviews'

export const movieReviewsSuccess = (reviews) => ({
    type: REVIEWS_SUCCESS,
    reviews
});

export const movieReviewsFaild = () => ({
    type: REVIEWS_FAILD, 
});

export const movieReviewsLoading = () => ({
    type: REVIEWS_LOADING,
});

export const movieReviewsFinish = () => ({
    type: REVIEWS_FINISH,
});

export const getMovieReviews = (idMovie, page) => (dispatch) => {
    dispatch(movieReviewsLoading());
    Api.getMovieReviews(idMovie, page)
    .then((json) => {
        dispatch(movieReviewsSuccess(json));
    })  
    .catch(err => {
        console.log(err);
        dispatch(movieReviewsFaild());
    })
  };

