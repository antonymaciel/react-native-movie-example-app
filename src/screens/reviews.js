import React from "react";
import { View, Text } from "react-native"
import styles from "../styles";

function Reviews(props) {
    return (<View style={styles.pageContainer}><Text>{props.msg}</Text></View>);
}

export default Reviews;