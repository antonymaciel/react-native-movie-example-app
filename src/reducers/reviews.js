import { REVIEWS_SUCCESS, REVIEWS_FAILD, REVIEWS_LOADING, REVIEWS_FINISH } from '../constants/reviews'

export const initialState = {
    reviews: null,
    totalPages: null, 
    totalReviews: null,
    loading: false,
    error: false
}


const reviews = (state = initialState, action) => {
    switch(action.type){
        case REVIEWS_SUCCESS:
            return {
                reviews: state.reviews ? [...state.reviews.results, ...action.reviews.results] : action.reviews.results,
                totalReviews: action.reviews.total_results,
                totalPages: action.reviews.total_pages,
                loading: false, 
                error: false
            };
        case REVIEWS_LOADING:
            return { 
                ...state, 
                loading: true,
                error: false
            };
        case REVIEWS_FAILD:
            return { 
                ...state, 
                loading: false, 
                error: true
            };
        case REVIEWS_FINISH:
            return initialState;    
        default:
                return state;
    } 
}

export default reviews;