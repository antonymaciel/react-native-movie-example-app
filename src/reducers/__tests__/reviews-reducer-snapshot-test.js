import reviews from '../reviews';
import { initialState } from '../reviews';
import { movieReviewsSuccess, movieReviewsFaild, movieReviewsLoading, movieReviewsFinish } from '../../actions/reviews';
import { reviews as data } from '../../../config/jest/mockData';

it('returns the same state on an unhandled action', () => {
  expect(reviews(initialState, {type: '_NULL'})).toMatchSnapshot();
});

it('handles load action', () => {
  expect(reviews(initialState, movieReviewsLoading())).toMatchSnapshot();
});

it('handles error action', () => {
  expect(reviews(initialState, movieReviewsFaild())).toMatchSnapshot();
});

it('handles finish action', () => {
expect(reviews(initialState, movieReviewsFinish())).toEqual({
    ...initialState,
    });
});

it('handles success print action', () => {
    expect(reviews(initialState, movieReviewsSuccess({total_pages: 15, results: data, total_results: 75}))).toMatchSnapshot();
});

