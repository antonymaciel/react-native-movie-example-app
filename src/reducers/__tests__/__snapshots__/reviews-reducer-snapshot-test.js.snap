// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles error action 1`] = `
Object {
  "error": true,
  "loading": false,
  "reviews": null,
  "totalPages": null,
  "totalReviews": null,
}
`;

exports[`handles load action 1`] = `
Object {
  "error": false,
  "loading": true,
  "reviews": null,
  "totalPages": null,
  "totalReviews": null,
}
`;

exports[`handles success print action 1`] = `
Object {
  "error": false,
  "loading": false,
  "reviews": Array [
    Object {
      "author": "Antony Maciel",
      "content": "Great Movie!!!",
      "id": 1,
      "url": "https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df",
    },
    Object {
      "author": "Antony Maciel",
      "content": "Great Movie!, always has amazing content 
 I love this movie",
      "id": 2,
      "url": "https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df",
    },
    Object {
      "author": "Antony Maciel",
      "content": "&^%$#@!",
      "id": 3,
      "url": "https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df",
    },
  ],
  "totalPages": 75,
  "totalReviews": 15,
}
`;

exports[`returns the same state on an unhandled action 1`] = `
Object {
  "error": false,
  "loading": false,
  "reviews": null,
  "totalPages": null,
  "totalReviews": null,
}
`;
